pkg/rpcclient/wsclient.go

```diff
@@ -795,11 +795,6 @@ func (c *WSClient) ReceiveExecutions(flt *neorpc.ExecutionFilter, rcvr chan<- *s
 	}
 	params := []any{"transaction_executed"}
 	if flt != nil {
-		if flt.State != nil {
-			if *flt.State != "HALT" && *flt.State != "FAULT" {
-				return "", errors.New("bad state parameter")
-			}
-		}
```

Comment:
So instead of this removed check you have to add client-side IsValid check. Add this check to `(c *WSClient) performSubscription`. Change `Comparator` interface so that `Filter() any` returns NotificationFilter interface.